### DISCUSSION TITLE
Does it hear bats?

### DISCUSSION THREAD

**Discussion by pe0ter** [stas-sl/esphome-sound-level-meter#24] [Q&A]
I am running your unmodified sensor-community-example script using an ICS43434 mic for a few days now, comparing it to two different sensor.community dnms mics and I am impressed. Your solution matches the dnms-ICS43434 very close. The new dnms IM72D128 mic has a lower noise level so curve goes down a bit more but for the rest it matches too. 
One difference I noticed though, in the late evening I see a couple of peaks on your ESP32 sensor I do not see on the dnms sensors, could it be your sensor can hear bats flying by?
sensor.community ID's #87587 ESP32 ICS43434, #55020 DNMS ICS43434, #86901 DNMS IM72D128

**Comment by stas-sl** [comment]
That's interesting to know :)

I never really tried to compare with DNMS side by side as I don't have one, but I studied its source code and tried to implement it similarly. So it should match theoretically, but that's good to know it matches in practice. Though, of course it also depends on hardware: mic, board, power supply, casing, etc...

As for bats... Don't know what to say ) We don't have them a lot in our area, and TBH I'm not sure whether those mics are capable of capturing beyond specified range of 20Hz-20kHz. Though if you are using same mics, they both should capture similar frequencies. However there could be small differences how A-weighting is implemented for frequencies >20kHz. Basically if your know the idea of A-weighting or read wikipedia, it is designed to match how humans perceive frequencies, so it gives more weight to 200Hz-10kHz range, and cuts off lower or higher frequencies. Another limitation is sampling rate of 48kHz which also makes capturing/processing of sounds with >24kHz impossible.
